src/components/createCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useState } from 'react';
+import Button from '../../common/Button';
+import './CreateCourse.css';
+import { CreateCourseProps } from './CreateCourse.types';
+import { getCourseDuration } from '../../helpers/getCourseDuration';
+
+const CreateCourse: React.FC<CreateCourseProps> = ({
+  setCoursesList,
+  setAuthorsList,
+  setCreateMode,
+  authorsList,
+}) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [courseAuthors, setCourseAuthors] = useState([]);
+  const [newAuthor, setNewAuthor] = useState('');
+  const [duration, setDuration] = useState(null);
+
+  const [errors, setErrors] = useState({
+    title: false,
+    description: false,
+    duration: false,
+    courseAuthors: false,
+    newAuthor: false,
+  });
+
+  const handleTitle = (event) => {
+    setErrors((prev) => ({
+      ...prev,
+      title: false,
+    }));
+    setTitle(event.target.value);
+  };
+
+  const handleDescription = (event) => {
+    setErrors((prev) => ({
+      ...prev,
+      description: false,
+    }));
+    setDescription(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    let hasErrors = false;
+    if (!title || title.length < 2) {
+      setErrors((prev) => ({
+        ...prev,
+        title: true,
+      }));
+      hasErrors = true;
+    }
+    if (!description || description.length < 2) {
+      setErrors((prev) => ({
+        ...prev,
+        description: true,
+      }));
+      hasErrors = true;
+    }
+
+    if (!duration) {
+      setErrors((prev) => ({
+        ...prev,
+        duration: true,
+      }));
+      hasErrors = true;
+    }
+
+    if (!courseAuthors.length) {
+      setErrors((prev) => ({
+        ...prev,
+        courseAuthors: true,
+      }));
+      hasErrors = true;
+    }
+
+    if (hasErrors) {
+      alert('Please, fill in all fields');
+      return;
+    }
+
+    const courseAuthorsId: string[] = [];
+    courseAuthors.forEach((auth) => {
+      return courseAuthorsId.push(auth.id);
+    });
+    const today = new Date('03/25/2015');
+
+    setCoursesList((prev) => {
+      const newCourseItem = {
+        id: Date.now(),
+        title,
+        description,
+        creationDate: today.toLocaleDateString(),
+        duration,
+        authors: courseAuthorsId,
+      };
+
+      return prev.concat(newCourseItem);
+    });
+    setCreateMode(false);
+  };
+
+  const handleAddAuthors = (author) => {
+    setCourseAuthors((prev) => {
+      if (!prev.find((aut) => aut.name === author.name)) {
+        return prev.concat(author);
+      } else {
+        return prev;
+      }
```

Comment:
You can use ternary operator for such constructions. But it's up to you of course.